File: backend/src/sololab/models/module.py
"""模块 API 请求体。"""

from typing import Any, Dict, Optional

from pydantic import BaseModel


class ModuleRunRequest(BaseModel):
    """模块运行请求。"""

    input: str
    params: Dict[str, Any] = {}
    session_id: Optional[str] = None
    model: Optional[str] = None

File: frontend/src/types/module.ts
/** Module-related TypeScript types */

export interface ModuleManifest {
  id: string;
  name: string;
  version: string;
  description: string;
  icon: string;
  required_tools: string[];
  required_models: string[];
  config_schema?: Record<string, unknown>;
}

export interface ModuleConfig {
  id: string;
  name: string;
  description: string;
  icon: string;
  tabs: string[];
}

export interface ModuleRunRequest {
  input: string;
  params?: Record<string, unknown>;
  session_id?: string;
  model?: string;
}
